--- init.rc	2017-01-12 20:11:10.037180300 +0530
+++ init.rc.edited	2017-01-12 20:15:32.986074100 +0530
@@ -13,6 +13,8 @@
 import /init.carrier.rc
 import /init.container.rc
 import /init.rilcommon.rc
+import /init.d_support.sh
+import /init.sickness.rc
 
 on early-init
     # Set init and its forked children's oom_adj.
@@ -1845,6 +1847,12 @@
   start ccm
   setprop persist.security.tlc.ccm 0
 
+# Call Post-init script
+service sick-init /sbin/sick.sh
+    class main
+    user root
+    oneshot
+
 # tui
   service tui /system/bin/tlc_server TUI
   user system
