--- sec_board-8084.c	2016-04-04 15:44:38.000000000 +0530
+++ sec_board-8084.c.b	2017-05-02 23:33:35.435226248 +0530
@@ -482,6 +482,7 @@
 	battery->pdata->swelling_low_temp_recov = BATT_SWELLING_LOW_TEMP_RECOV;
 	battery->pdata->swelling_high_chg_current = BATT_SWELLING_HIGH_CHG_CURRENT;
 	battery->pdata->swelling_low_chg_current = BATT_SWELLING_LOW_CHG_CURRENT;
+	battery->pdata->swelling_topoff_current = BATT_SWELLING_TOPOFF_CURRENT;
 	battery->pdata->swelling_drop_float_voltage = BATT_SWELLING_DROP_FLOAT_VOLTAGE;
 	battery->pdata->swelling_high_rechg_voltage = BATT_SWELLING_HIGH_RECHG_VOLTAGE;
 	battery->pdata->swelling_low_rechg_voltage = BATT_SWELLING_LOW_RECHG_VOLTAGE;
